aquarius/events/nft_ownership.py

```diff
@@ -0,0 +1,119 @@
+                logging.error("Connection to ES failed. Trying to connect to back...")
+                time.sleep(5)
+            # logging.info("Stable connection to ES.")
+            last_block_record = self._es_instance.es.get(
+                index=self._db_index, id=self._index_name, doc_type="_doc"
+            )["_source"]
+            block = (
+                last_block_record["last_block"]
+                if last_block_record["last_block"] >= 0
+                else get_defined_block(self._chain_id)
+            )
+        except Exception as e:
+            # Retrieve the defined block.
+            if type(e) == elasticsearch.NotFoundError:
+                block = get_defined_block(self._chain_id)
+                logger.info(f"Retrieved the default block. NotFound error occurred.")
+            else:
+                logging.error(f"Cannot get last_block error={e}")
+        return block
+
+    def store_last_processed_block(self, block):
+        """Stores last processed block
+
+        Args:
+            block: last block that was processed
+        """
+        # make sure that we don't write a block < then needed
+        stored_block = self.get_last_processed_block()
+        logger.info(f"Storing last_processed_block {block}  (In Es: {stored_block})")
+        if block <= stored_block:
+            return
+        record = {"last_block": block}
+        try:
+            self._es_instance.es.index(
+                index=self._db_index,
+                id=self._index_name,
+                body=record,
+                doc_type="_doc",
+                refresh="wait_for",
+            )["_id"]
+
+        except elasticsearch.exceptions.RequestError:
+            logger.error(
+                f"store_last_processed_block: block={block} type={type(block)}, ES RequestError"
+            )
+
+    def update_lists(self):
+        """
+        Grab and process all nft transfership events from subgraph, starting from last known block
+        """
+        start_block = self.get_last_processed_block()
+        # never go past last indexed block, because we will not have the ddos
+        end_block = self._events_monitor.get_last_processed_block()
+        if end_block > start_block:
+            nft_transfers_list = get_nft_transfers(
+                start_block, end_block, self._chain_id
+            )
+            for transfer in nft_transfers_list:
+                did = make_did(
+                    Web3.toChecksumAddress(transfer["nft"]["id"]), self._chain_id
+                )
+                try:
+                    asset = self._es_instance.read(did)
+                    asset["nft"]["owner"] = Web3.toChecksumAddress(
+                        transfer["newOwner"]["id"]
+                    )
+                    self._es_instance.update(asset, did)
+                    logger.debug(f"Updated {did}: new owner: {asset['nft']['owner']}")
+                except NotFoundError:
```

Comment:
you can also catch both errors, i.e. except NotFoundError, Exception as e. Then only modify the "Not found" / "{e}" part of the message to help DRY-ness.

Reply:
I do wonder if it fails with something else than NotFoundError then should we have a retry queue handler?